app/celery/broadcast_message_tasks.py

```diff
@@ -78,6 +87,11 @@ def send_broadcast_provider_message(broadcast_event_id, provider):
 @notify_celery.task(name='trigger-link-test')
 def trigger_link_test(provider):
     identifier = str(uuid.uuid4())
+    formatted_seq_number = None
+    if provider == BroadcastProvider.VODAFONE:
+        sequence = Sequence('broadcast_provider_message_number_seq')
+        sequential_number = db.session.connection().execute(sequence)
+        formatted_seq_number = format_sequential_number(sequential_number)
```

Comment:
i think i'd rather have this kind of logic inside the `CBCProxyVodafone` class. If every provider has a different if block in here and in the `broadcast_provider_dao.py::create_broadcast_provider_message` function as well,  it might get a bit messy, and that's exactly what the different cbc_proxy_clients are there to avoid.

However, that means putting DB stuff inside the cbc_proxy.py file, and that's not consistent with how we've done things so far, so that makes me a bit hesitant. Happy to hear other thoughts from the rest of the dev team 🤔

Reply:
agree this is a bit weird but not a big issue right now. keep an eye on it if it grows as we add more providers/fields/etc and we might want to move it later if necessary

Reply:
As per our chat, let's leave it here now to try and avoid putting db stuff in the clients, but if it becomes unwieldy as more providers are added, we will look to refactor it then.